pkg/server/unit.go

```diff
@@ -926,19 +929,30 @@ func (u *Unit) Run(podname, hostname string, command []string, workingdir string
 	if rootfs != "" {
 		oldrootfs := fmt.Sprintf("%s/.oldrootfs", rootfs)
 
-		// We used to bind mount rootfs so we could change it to private for
-		// pivot_root. Now that the rootfs is an overlayfs mount, this is not
-		// needed, and we can change the overlayfs mount to private directly.
-		//if err := mounter.BindMount(rootfs, rootfs); err != nil {
-		//	glog.Errorf("Mount() %s: %v", rootfs, err)
-		//	u.setStateToStartFailure(err)
-		//	return err
-		//}
-		privFlags := uintptr(syscall.MS_PRIVATE)
-		if err := mount.ShareMount(rootfs, privFlags); err != nil {
-			glog.Errorf("ShareMount(%s, private): %v", rootfs, err)
-			u.setStateToStartFailure(err)
-			return err
+		// Now that we allow both overlayfs and direct extraction we cannot
+		// assume the below mounting permissions. We need to decide whether
+		// we need to use bind mount for the direct layer extraction or
+		// set mount permissions for "/" and "rootfs" if using overlayfs
+		useOverlayfs := u.unitConfig.UseOverlayfs
+		if useOverlayfs {
```

Comment:
nitpicking (feel free to ignore): I like to avoid `if / else` if possible, and do:
```
if !notDefaultBehavior:
     notDefaultBehavior logic
defaultBehavior logic
```

Reply:
We need the else in this case due to the fact that only one of the two cases can be executed. If we are not using the overlayfs we only need to bind mount. If we are using overlayfs we set the rootfs flags to private then set the parent (in this case "/") to be private as well.

Otherwise i completely agree with your thoughts surrounding `if/else`